backend/app/modules/snapshots/finalized/core.py

```diff
@@ -9,24 +10,46 @@
     AllocationDTO,
     AccountFundsDTO,
 )
+from app.engine.projects.rewards import ProjectRewardDTO
 
 FinalizedProjectRewards = namedtuple(
     "FinalizedProjectRewards", ["rewards", "rewards_sum"]
 )
 
 
-def calculate_leftover(octant_rewards: OctantRewardsDTO, total_withdrawals: int):
+def _calculate_octant_rewards_leftover(
+    octant_rewards: OctantRewardsDTO, total_withdrawals: int
+) -> int:
     return (
         octant_rewards.staking_proceeds
         - octant_rewards.operational_cost
         - total_withdrawals
     )
 
 
+def _calculate_matched_rewards_leftover(
+    project_rewards: List[ProjectRewardDTO], total_matched_rewards: int
+) -> int:
+    used_matched_rewards = sum(r.matched for r in project_rewards)
+    return total_matched_rewards - used_matched_rewards
+
+
+def calculate_leftover(
```

Comment:
@leoni-q I was thinking about making a separate service for leftovers but finally I decided to keep it as a core logic since it's very likely to me now to always save the unused Matched Rewards in the leftovers, not only in the epoch4 configuration (the fact is that summa summarum we'll be deciding manually whether to use MR from leftovers or not). But if you don't agree, let me know, we'll figure this out.

Reply:
ok, it can stay here